### PR TITLE
Add Action State Button to the StoryItem

### PR DESCRIPTION
**Pull request by negreirosleo**
The action state button was hardcoded, but now it changes according
to the story state.

### CODE REVIEW

File: spec/javascripts/components/story/story_actions_spec.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+describe('<StateActions />', () => {
+  describe('When estimate is null',() => {
+    it("renders <StoryPoints /> component", () => {
+      const props = {
+        storyType: "feature",
+        estimate: null,
+        state: "started"
+      };
+      const wrapper = shallow(<StateActions {...props} />);
+      expect(wrapper.find(StoryPoints)).toHaveLength(1);
```
Comment (brunohkbx) [inline spec/javascripts/components/story/story_actions_spec.js]: `expect(wrapper.find(StoryPoints)).toExist();`

Comment (negreirosleo) [reply spec/javascripts/components/story/story_actions_spec.js]: Vou ajeitar, valheu

File: spec/javascripts/components/story/story_actions_spec.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      "accepted",
+      "unstarted"
+    ];
+    states.forEach((state) => {
+      describe(`When state = ${state}`, () => {
+        it('renders the <StateButton /> component',() => {
+          const wrapper = shallow(<StateActions state={state} estimate={1} storyType="feature" />);
+          const actions = StoryActionFor[state]
+          actions.forEach((action) => {
+            expect(wrapper.find(`StateButton[action="${action}"]`).exists()).toEqual(true);
```
Comment (brunohkbx) [inline spec/javascripts/components/story/story_actions_spec.js]: I think we can also improve this spec :thinking: 

```suggestion
            expect(wrapper.find(`StateButton[action="${action}"]`)).toExist();
```

Comment (negreirosleo) [reply spec/javascripts/components/story/story_actions_spec.js]: Spec improved.

File: app/assets/javascripts/components/story/StoryItem.js [hunk 3]
```diff
@@ -11,33 +11,51 @@ import {
...
       isStoryNotEstimated(storyType, estimate)
         ? <StoryPoints />
-        : <ButtonStart />
+        : StoryActionFor(state).map((stateAction) => <StateButton action={stateAction} key={stateAction} />)
     }
   </div>
-)
+);
+
+export const StoryActionFor = (state) => StateAction[state] || StateAction['unstarted'];
```
Comment (murilovarela) [inline app/assets/javascripts/components/story/StoryItem.js]: StateAction.unstarted

File: app/assets/javascripts/components/story/StoryItem.js [hunk 4]
```diff
@@ -11,33 +11,51 @@ import {
...
+);
+
+export const StoryActionFor = (state) => StateAction[state] || StateAction['unstarted'];  
+
 StateActions.propTypes = {
-  storyType : PropTypes.string.isRequired
-}
+  storyType: PropTypes.string.isRequired,
+  estimate: PropTypes.number,
+  state: PropTypes.string
```
Comment (murilovarela) [inline app/assets/javascripts/components/story/StoryItem.js]: Falta o defaultProps do StateActions

**Pull request merged**